core/transactions/transactions.cxx

```diff
@@ -174,19 +174,11 @@ wrap_public_api_run(transactions& txns,
 {
   return wrap_run(txns, config, max_attempts, [fn = std::forward<Handler>(fn)](const auto& ctx) {
     const couchbase::error err = fn(ctx);
-    if (err) {
-      if (err.ec() == errc::transaction_op::transaction_op_failed) {
-        if (auto cause = err.cause(); cause) {
-          throw transaction_operation_failed(FAIL_OTHER, err.message())
-            .cause(external_exception_from_transaction_op_errc(
-              errc::transaction_op(cause->ec().value())));
-        }
-      } else {
-        if (err.ec().category() == core::impl::transaction_op_category()) {
-          throw op_exception(err);
-        }
-        throw std::runtime_error(err.ec().message());
+    if (err && err.ec() != errc::transaction_op::transaction_op_failed) {
+      if (err.ec().category() == core::impl::transaction_op_category()) {
+        throw op_exception(err);
       }
+      throw std::runtime_error(err.ec().message());
```

Comment:
how about throwing `std::system_error(err.ec())` it is still generic, but its constructor accepts `std::error_code`
https://en.cppreference.com/w/cpp/error/system_error/system_error
```
system_error( std::error_code ec );
```

Reply:
Done